capplets/time-admin/src/main.c

```diff
@@ -0,0 +1,430 @@
+    }   
+    
+    return Pixbuf;
+}   
+static gboolean on_window_quit (GtkWidget *widget,
+                                GdkEvent  *event,
+                                gpointer   user_data)
+{
+    TimeAdmin *ta = (TimeAdmin *)user_data;
+    
+    QuitApp(ta);
+    return TRUE;
+}
+static void CloseWindow (GtkButton *button,gpointer data)
+{
+    TimeAdmin *ta = (TimeAdmin *)data;
+
+    QuitApp(ta);
+}    
+static void InitMainWindow(TimeAdmin *ta)
+{
+    GtkWidget *Window;
+    GdkPixbuf *AppIcon;
+    //GError    *error = NULL;
+
+    Window = gtk_window_new(GTK_WINDOW_TOPLEVEL);
+    ta->MainWindow = WindowLogin = Window;
+    gtk_window_set_deletable(GTK_WINDOW(Window),FALSE);
+    gtk_window_set_hide_titlebar_when_maximized(GTK_WINDOW(Window),TRUE);
+    gtk_window_set_position(GTK_WINDOW(Window), GTK_WIN_POS_CENTER);
+    gtk_window_set_title(GTK_WINDOW(Window),_("Time and Date Manager")); 
+    gtk_container_set_border_width(GTK_CONTAINER(Window),10);
+    gtk_widget_set_size_request(Window, 300, 360);
+    g_signal_connect(G_OBJECT(Window),
+                    "delete-event",
+                     G_CALLBACK(on_window_quit),
+                     ta);
+
+   /* 
+    ua->Permission = polkit_permission_new_sync (USER_ADMIN_PERMISSION, NULL, NULL, &error);
+    ua->ButtonLock = gtk_lock_button_new(ua->Permission);
+    gtk_lock_button_set_permission(GTK_LOCK_BUTTON (ua->ButtonLock),ua->Permission);
+    g_signal_connect(ua->Permission, 
+                    "notify",
+                     G_CALLBACK (on_permission_changed), 
+                     ua);
+  */  
+    AppIcon = GetAppIcon();
+    if(AppIcon)
+    {
+        gtk_window_set_icon(GTK_WINDOW(Window),AppIcon);
+        g_object_unref(AppIcon);
+    }   
+}
+
+static int RecordPid(void)
+{
+    int pid = 0;
+    int fd;
+    int Length = 0; 
+    char WriteBuf[30] = { 0 };
+    fd = open(LOCKFILE,O_WRONLY|O_CREAT|O_TRUNC,0777);
+    if(fd < 0)
+    {
+         MessageReport(_("open file"),_("Create pid file failed"),ERROR);
+         return -1;      
+    }       
+    chmod(LOCKFILE,0777); 
+    pid = getpid();
+    sprintf(WriteBuf,"%d",pid);
+    Length = write(fd,WriteBuf,strlen(WriteBuf));
+    if(Length <= 0 )
+    {
+        MessageReport(_("write file"),_("write pid file failed"),ERROR);
+        return -1;      
+    }			
+    close(fd);
+
+    return 0;
+}        
+/******************************************************************************
+* Function:              ProcessRuning      
+*        
+* Explain: Check whether the process has been started,If the process is not started, 
+*          record the current process ID =====>"/tmp/user-admin.pid"
+*        
+* Input:         
+*        
+*        
+* Output:  start        :TRUE
+*          not start    :FALSE
+*        
+* Author:  zhuyaliang  31/07/2018
+******************************************************************************/
+static gboolean ProcessRuning(void)
+{
+    int fd = 0;
```

Comment:
**cppcheck warning**
Id: unreadVariable
Variable 'fd' is assigned a value that is never used.

Reply:
Can't this be initialized to zero?

Reply:
fd initialization is overridden in line 166, and the initialization value (0) is not used:
```C
fd = open(LOCKFILE,O_RDONLY);
```
It's a cppcheck warning, we suggest to developers to keep [build, cppcheck and clang reports](https://mate-control-center.mate-desktop.dev/) as clean as possible.

Reply:
I will try to get rid of these warnings